From eb128daaf836c0dd8a2d01ecac84dc178fce5759 Mon Sep 17 00:00:00 2001
From: DEREFERENC3D <dereferenc3d@protonmail.com>
Date: Mon, 5 May 2025 21:11:04 +0200
Subject: [PATCH 1/1] docker: Fix permission issues

* Bind to port 8080 instead of 80 - not required in new versions of
  Docker, but still required e.g. in Podman
* Set the permissions on nginx's "html" directory - no idea why the
  "--chmod" on the COPY does not work here
---
 Dockerfile        | 3 +++
 docker/nginx.conf | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/Dockerfile b/Dockerfile
index 804d0a23..40ebde0f 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -23,6 +23,9 @@ RUN --mount=type=cache,target=/root/.local/share/pnpm \
 FROM nginxinc/nginx-unprivileged:alpine
 
 COPY --chown=101:101 --from=build /app/dist/ /usr/share/nginx/html/
+USER root
+RUN chown 101:101 /usr/share/nginx/html
+USER 101
 
 COPY --chown=101:101 docker/nginx.conf /etc/nginx/conf.d/default.conf
 
diff --git a/docker/nginx.conf b/docker/nginx.conf
index 3b7ef86b..94ed4752 100644
--- a/docker/nginx.conf
+++ b/docker/nginx.conf
@@ -1,6 +1,6 @@
 server {
-    listen 80;
-    listen [::]:80;
+    listen 8080;
+    listen [::]:8080;
     server_name localhost;
     error_log off;
 
-- 
2.49.0

